rgb.lua

```diff
@@ -13,10 +23,29 @@ local digiline_rules = {
   {x =  0, y =  0,z = -2,},
 }
 
-local digiline_action = function(pos, node, channel, msg)
+local digiline_action = function(pos, node, channel, mixed_msg)
   local setchannel = minetest.get_meta(pos):get_string('channel')
   -- Ignore other digiline channels
-  if channel ~= setchannel or type(msg) ~= 'table' then return end
+  if channel ~= setchannel then return end
+  local msg_type = type(mixed_msg)
+  local msg = {}
+  if 'string' == msg_type then
+    local s = mixed_msg:lower()
+    if 'on' == s then
+      msg.switch = 1
+    elseif 'off' == s then
+      msg.switch = 0
+    elseif 'reset' == s then
+      msg.reset = 1
+    --elseif '#' == s:sub(1, 1) then
+      -- TODO:
+      --msg.color = translatedColour(s)
```

Comment:
this is easy to implement with `tonumber`, just feed it splitted hex values to get r, g and b.
```lua
s = "#a1b2c3" -- dec 161, 178, 195
r = tonumber("0x" .. s:sub(2,3))
g = tonumber("0x" .. s:sub(4,5))
b = tonumber("0x" .. s:sub(6,7))
```

Reply:
I didn't write this part, because Grid was already working on that. Thanks for the snippet anyway.

Reply:
@S-S-X, I think it's not that simple. 256-color palette, used by unifieddyes seem to be having HSV structure. It has 24 hues, 10 levels of value and I'm not sure how and if it uses saturation. The closest approach I see there is to convert the color like you do to find a hue, apply value and... Do something about the saturation level, as well.